engine/src/main/java/org/terasology/physics/bullet/BulletPhysics.java

```diff
@@ -710,72 +745,78 @@ private PairCachingGhostObject createCollider(Vector3f pos, ConvexShape shape, s
+            for (int x = 0; x < pairs.size(); x++) {
+                btBroadphasePair initialPair = pairs.at(x);
                 EntityRef otherEntity = null;
-                if (initialPair.pProxy0.clientObject == trigger) {
-                    if (((CollisionObject) initialPair.pProxy1.clientObject).getUserPointer() instanceof EntityRef) {
-                        otherEntity = (EntityRef) ((CollisionObject) initialPair.pProxy1.clientObject).getUserPointer();
+                btBroadphaseProxy p0 = btBroadphaseProxy.obtain(initialPair.getPProxy0().getCPointer(), false);
+                btBroadphaseProxy p1 = btBroadphaseProxy.obtain(initialPair.getPProxy1().getCPointer(), false);
+
+                if (p0.getClientObject() == trigger.getCPointer()) {
+
+                    btCollisionObject other = btCollisionObject.getInstance(p1.getClientObject());
+                    if (other.userData instanceof EntityRef) {
+                        otherEntity = (EntityRef) other.userData;
                     }
                 } else {
-                    if (((CollisionObject) initialPair.pProxy0.clientObject).getUserPointer() instanceof EntityRef) {
-                        otherEntity = (EntityRef) ((CollisionObject) initialPair.pProxy0.clientObject).getUserPointer();
+                    btCollisionObject other = btCollisionObject.getInstance(p0.getClientObject());
+                    if (other.userData instanceof EntityRef) {
+                        otherEntity = (EntityRef) other.userData;
                     }
                 }
                 if (otherEntity == null || otherEntity == EntityRef.NULL) {
                     continue;
                 }
-                BroadphasePair pair = world.getPairCache().findPair(initialPair.pProxy0, initialPair.pProxy1);
+                btBroadphasePair pair = discreteDynamicsWorld.getPairCache().findPair(p0, p1);
                 if (pair == null) {
                     continue;
                 }
+
                 manifolds.clear();
-                if (pair.algorithm != null) {
-                    pair.algorithm.getAllContactManifolds(manifolds);
+                if (pair.getAlgorithm() != null) {
+                    pair.getAlgorithm().getAllContactManifolds(manifolds);
                 }
-                for (PersistentManifold manifold : manifolds) {
+                for (int y = 0; y < manifolds.size(); y++) {
+
+                    btPersistentManifold manifold = manifolds.atConst(y);
                     for (int point = 0; point < manifold.getNumContacts(); ++point) {
-                        ManifoldPoint manifoldPoint = manifold.getContactPoint(point);
-                        if (manifoldPoint.getDistance() < 0) {
+                        btManifoldPoint manifoldPoint = manifold.getContactPoint(point);
+                        if (manifoldPoint.getDistance() < 0.f) {
+                            Vector3f a1 = new Vector3f();
```

Comment:
More meaningful variable names than `a1` 2 and 3? Seeing more, than those three as well